src/WebJobs.Script.WebHost/SecretManager.cs

```diff
@@ -101,6 +104,45 @@ public virtual FunctionSecrets GetFunctionSecrets(string functionName)
             });
         }
 
+        /// <summary>
+        /// Iterate through all function secret files and remove any that don't correspond
+        /// to a function.
+        /// </summary>
+        /// <param name="rootScriptPath">The root function directory.</param>
+        public void PurgeOldFiles(string rootScriptPath)
+        {
+            if (!Directory.Exists(rootScriptPath))
+            {
+                return;
+            }
+
+            var functionLookup = Directory.EnumerateDirectories(rootScriptPath).ToLookup(p => Path.GetFileName(p), StringComparer.OrdinalIgnoreCase);
+            var secretsDirectory = new DirectoryInfo(_secretsPath);
+
+            foreach (var secretFile in secretsDirectory.GetFiles("*.json"))
+            {
+                string fileName = Path.GetFileNameWithoutExtension(secretFile.Name);
+                if (string.Compare(fileName, "host") == 0)
```

Comment:
As an aside, I'm worried that this 'host' special cases is going to hurt us. e.g. what happens if someone tries to create a function called host?

At the very least we should use a helper to test for it, since we have same test in other places (like https://github.com/Azure/azure-webjobs-sdk-script/blob/560deb037084cd9f30ee59338e43f9f1a4a17f56/src/WebJobs.Script.WebHost/SecretManager.cs#L123).